### ISSUE TITLE
Built-in: importstr/yaml

### ISSUE THREAD

**Issue by steeling** [kcl-lang/kcl#995]
## Feature Request

**Is your feature request related to a problem? Please describe:**
I want to declaritively create text files and import them into kcl.

**Describe the feature you'd like:**
I'd like to create configmaps through kcl, ie:

```python
kind: "ConfigMap"
apiVersion: "v1"
data: {
   "myfile": importstr(myfile.txt)
}
```
**Describe alternatives you've considered:**
Create generative rules outside of kcl

Other improvements might be:

`importYaml`, which would be able to import a yaml file, and convert to a KCL object.
`glob` to allow creating a list of filenames, that I can then iterate over to create a set of files, ie:

```python
kind: "ConfigMap"
apiVersion: "v1"
data: {
   filename: importstr(filename) for filename in glob('mydata/*.yaml')
}
```
This feature does exist in jsonnet

**Comment by Peefy** [comment]
Thank you for your feedback ❤️. Considering that corresponding IO functions or configmap generators are provided in many projects such as Kustomize, Helm, Terraform, Jsonnet, etc., this feature is considered suitable to be added as functions in the `io` system modules in version 0.8 of KCL just most of what is done in this issue https://github.com/kcl-lang/kcl/issues/313

**Comment by shivam-Purohit** [comment]
hey @Peefy I wanted to start with the project, can i be any help in this?

**Comment by Peefy** [comment]
> hey @Peefy I wanted to start with the project, can i be any help in this?

Yes, you can open a PR to impl the `io` package for KCL.

**Comment by shivam-Purohit** [comment]
> Yes, you can open a PR to impl the `io` package for KCL.

I will look into it and come up with a draft pr.

**Comment by Peefy** [comment]
Whether this feature is Implemented in the form of a `read_file` function or `importstr`/`importyaml`, this feature must be additionally added with the sandbox feature to ensure security. The default sandbox workspace is `workdir`, and we need to discuss the relevant design here. cc @shivam-Purohit @zong-zhe

**Comment by shivam-Purohit** [comment]
@Peefy is this on high priority? sorry, but I am currently a bit busy. I am unassigning myself for now, if someone is interested they can try freely, it's up for grabs. I will try this in few weeks. Is that okay?

**Comment by Peefy** [comment]
Related Issue: #1049 
Closed by #1034, #1054, #1056, #1051 and #1052.
